redfish-core/lib/environment_metrics.hpp

```diff
@@ -1,18 +1,140 @@
 #pragma once
 
 #include "app.hpp"
+#include "dbus_utility.hpp"
+#include "error_messages.hpp"
 #include "query.hpp"
 #include "registries/privilege_registry.hpp"
+#include "str_utility.hpp"
 #include "utils/chassis_utils.hpp"
+#include "utils/fan_utils.hpp"
 
 #include <boost/url/format.hpp>
 
+#include <array>
+#include <cstdint>
+#include <functional>
 #include <memory>
 #include <optional>
 #include <string>
+#include <string_view>
 
 namespace redfish
 {
+inline void
+    updateFanSensorList(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                        const std::string& chassisId,
+                        const std::string& fanSensorPath, double value)
+{
+    std::string fanSensorName =
+        sdbusplus::message::object_path(fanSensorPath).filename();
+    if (fanSensorName.empty())
+    {
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    nlohmann::json::object_t item;
+    item["DataSourceUri"] = boost::urls::format(
+        "/redfish/v1/Chassis/{}/Sensors/{}", chassisId, fanSensorName);
+    item["DeviceName"] = "Chassis #" + fanSensorName;
+    item["SpeedRPM"] = value;
+
+    nlohmann::json& fanSensorList =
+        asyncResp->res.jsonValue["FanSpeedsPercent"];
+    fanSensorList.emplace_back(std::move(item));
+    std::sort(fanSensorList.begin(), fanSensorList.end(),
```

Comment:
I don't think - the use of td::vector::insert()` is faster.
To do it (on vector), it first needs to find the location and the insert it -- which may potentially re-allocate / move all successive entries.  Then, its cost will be  `O(N^2)`.
- `O(n)` (for find) + `O(n)` [for insert) - for each element.   (If insert is done at the end, it would become similar to the current approach).
- Not only the computational complexity, it could cause the unnecessary memory movement.

The current approach will be `O(N*log(N)` (i.e. for sort mostly).

In general, I don't think there will be many fan sensors for each fan, and thus the differences of the cost complexity can be negligible...and then the simpler code may be better to use.